worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,373 @@
+
+        if self.options.goal.value == 1:
+            self.multiworld.get_location("Reconstituted Luigi", self.player).place_locked_item(self.create_item("Saved Baby Luigi"))
+        else:
+            self.multiworld.get_location("King Bowser's Castle: Level Clear", self.player).place_locked_item(self.create_item("Saved Baby Luigi"))
+
+        self.multiworld.get_location("Touch Fuzzy Get Dizzy: Gather Coins", self.player).place_locked_item(self.create_item("Bandit Consumables"))
+        self.multiworld.get_location("The Cave Of the Mystery Maze: Seed Spitting Contest", self.player).place_locked_item(self.create_item("Bandit Watermelons"))
+        self.multiworld.get_location("Lakitu's Wall: Gather Coins", self.player).place_locked_item(self.create_item("Bandit Consumables"))
+        self.multiworld.get_location("Ride Like The Wind: Gather Coins", self.player).place_locked_item(self.create_item("Bandit Consumables"))
+
+    def generate_early(self) -> None:
+        setup_gamevars(self)
+
+
+    def get_excluded_items(self) -> Set[str]:
+        excluded_items: Set[str] = set()
+
+        starting_gate = ["World 1 Gate", "World 2 Gate", "World 3 Gate",
+                         "World 4 Gate", "World 5 Gate", "World 6 Gate"]
+
+
+        excluded_items.add(starting_gate[self.options.starting_world])
+
+        if not self.options.shuffle_midrings:
+            excluded_items.add('Middle Ring')
+
+        if not self.options.add_secretlens:
+            excluded_items.add('Secret Lens')
+
+        if not self.options.extras_enabled:
+            excluded_items.add('Extra Panels')
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.options.split_extras:
+            excluded_items.add('Extra Panels')
+        else:
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.options.split_bonus:
+            excluded_items.add('Bonus Panels')
+        else:
+            excluded_items.add('Bonus 1')
+            excluded_items.add('Bonus 2')
+            excluded_items.add('Bonus 3')
+            excluded_items.add('Bonus 4')
+            excluded_items.add('Bonus 5')
+            excluded_items.add('Bonus 6')
+
+        return excluded_items
+
+    def create_item_with_correct_settings(self, player: int, name: str) -> Item:
+        data = item_table[name]
+        item = Item(name, data.classification, data.code, player)
+
+        if not item.advancement:
+            return item
+
+        if name == 'Car Morph' and not self.options.stage_logic:
```

Comment:
```suggestion
        if name == "Car Morph" and self.options.stage_logic != StageLogic.option_strict:
```
Please make this fix in addition to accepting the PR.